### PR TITLE
GEODE-8953: Re-introduce transaction details regarding non-transactional changes

### PR DESCRIPTION
**Pull request by davebarnes97**
Restore some cautionary text that was removed during an earlier re-write of the transaction section.

### CODE REVIEW

**Review by albertogpz — APPROVED** [review]
Looks good to me. Thanks for addressing this.

File: geode-docs/developing/transactions/design_considerations.html.md.erb [hunk 1]
```diff
@@ -117,6 +118,21 @@ immediately after the commit.
 A transaction disables expiration on any region entries affected
 by the transaction.
 
+## <a id="transactions-nontransactions" class="no-quick-link"></a>Mixing Transactions with Nontransactional Operations
```
Comment (karensmolermiller) [inline geode-docs/developing/transactions/design_considerations.html.md.erb:121]: Add hyphen to "Nontransactional" to become Non-transactional.

File: geode-docs/developing/transactions/design_considerations.html.md.erb [hunk 2]
```diff
@@ -117,6 +118,21 @@ immediately after the commit.
...
+
+For best performance, non-transactional operations do not acquire the exclusive locks used to check
+for conflicts in a transaction. A transaction operating on the same data as a non-transactional actor
+is unable to detect the conflict caused by a non-transactional operation.
+
+A user application should adopt the policy of always using transactions or no transactions at all
```
Comment (karensmolermiller) [inline geode-docs/developing/transactions/design_considerations.html.md.erb:127]: The phrase "no transactions at all" feels awkward to me. Maybe better (?):
> A user application should adopt the policy of always using transactions or the policy of not using transactions at all for certain regions or sets of entries, so transactional entries will not be modified by operations
outside of the transaction.

**Review by karensmolermiller — CHANGES_REQUESTED** [review]

File: geode-docs/developing/transactions/design_considerations.html.md.erb (hunk 1 above)
Comment (davebarnes97) [reply geode-docs/developing/transactions/design_considerations.html.md.erb:121]: Good catch, thanks.

File: geode-docs/developing/transactions/design_considerations.html.md.erb (hunk 2 above)
Comment (davebarnes97) [reply geode-docs/developing/transactions/design_considerations.html.md.erb:127]: I've replaced this paragraph with:
If using transactions, an application should adopt the policy of designating certain regions or sets of entries
exclusively for transactional puts, updates, and deletions, so transactional entries will not be modified by non-transactional.
operations.

**Review by karensmolermiller — APPROVED** [review]

**Pull request merged**